xc/xenops_server_xen.ml

```diff
@@ -2194,12 +2199,12 @@ module VBD = struct
              let qemu_domid = Opt.default (this_domid ~xs) (get_stubdom ~xs frontend_domid) in
              let qemu_frontend = match Device_number.spec device_number with
                | Device_number.Ide, n, _ when n < 4 ->
+                 let index = Device_number.to_disk_number device_number in
                  begin match vbd.Vbd.backend with
```

Comment:
Why not
```ocaml
let bd =
    vbd.Vbd.backend
    |> Opt.map (fun _ -> create_vbd_frontend ~xc ~xs task qemu_domid vdi)
in
Some (index, bd)
```
It would save you the `begin..end` block